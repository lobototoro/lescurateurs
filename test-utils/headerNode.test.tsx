import HeaderMenu from '@/app/editor/components/headerMenu';
import { render } from '@testing-library/react';
import { expect, it, vi } from 'vitest';

it('Should render correctly', () => {
  const role = 'admin';
  const permissions = JSON.stringify(['read:articles', 'edit:articles']);
  const setSelection = vi.fn();
  const { asFragment } = render(<HeaderMenu role={role} permissions={permissions} setSelection={setSelection} selection='' />);
  expect(asFragment()).toMatchSnapshot();
});