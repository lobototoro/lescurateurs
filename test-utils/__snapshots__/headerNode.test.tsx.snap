// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Should render correctly 1`] = `
<DocumentFragment>
  <div>
    <h2
      class="undefined article-title_large article-title_black mt-5 mb-5"
    >
      Ã‰DITEUR
    </h2>
  </div>
  <nav
    aria-label="main navigation"
    class="navbar box"
    role="navigation"
  >
    <div
      class="navbar-brand"
    >
      <a
        aria-expanded="false"
        aria-label="menu"
        class="navbar-burger"
        data-target="navbarMenuItems"
        role="button"
      >
        <span
          aria-hidden="true"
        />
        <span
          aria-hidden="true"
        />
        <span
          aria-hidden="true"
        />
        <span
          aria-hidden="true"
        />
      </a>
    </div>
    <div
      class="navbar-menu is-flex-direction-column"
      id="navbarMenuItems"
    >
      <div
        class="navbar-start container is-flex-wrap-wrap"
      >
        <div
          class="navbar-item"
        >
          <a
            class="button "
          >
            manage user
          </a>
        </div>
        <div
          class="navbar-item"
        >
          <a
            class="button "
          >
            edit articles
          </a>
        </div>
      </div>
      <div
        class=""
      >
        <div
          class="navbar-item"
        >
          <div
            class="buttons"
          >
            <a
              class="button is-info"
              href="/auth/logout"
            >
              Log out
            </a>
            <span
              class="tag is-dark"
            >
              <strong>
                Role : admin
              </strong>
            </span>
          </div>
        </div>
      </div>
    </div>
    <hr />
  </nav>
</DocumentFragment>
`;
